server/src/test/java/org/elasticsearch/index/engine/CombinedDeletionPolicyTests.java

```diff
@@ -211,6 +215,31 @@ public void testDeleteInvalidCommits() throws Exception {
         }
     }
 
+    public void testKeepOnlyStartingCommitOnInit() throws Exception {
```

Comment:
can you add java docs explaining why this behavior is needed. I think it will be hard to figure out.